python/cudf_polars/cudf_polars/experimental/rapidsmpf/core.py

```
@@ -99,13 +70,32 @@ def evaluate_logical_plan(
     -------
     The output DataFrame and metadata collector.
     """
-    query_id = uuid.uuid4()
+    # For default_singleton, inject the process-wide DefaultSingletonEngine instance
+    # into config_options before treating it as a regular SPMDEngine.
+    if config_options.executor.cluster == "default_singleton":
+        from cudf_polars.experimental.rapidsmpf.frontend.default_singleton_engine import (
+            DefaultSingletonEngine,
+        )
```

Ruff (0.15.12)

[warning] `import` should be at the top-level of a file

(PLC0415)